src/Elastic.Apm/Model/Span.cs

```diff
@@ -135,14 +141,25 @@ internal Span StartSpanInternal(string name, string type, string subType = null,
 
 			if (!string.IsNullOrEmpty(action)) retVal.Action = action;
 
-			_logger.Debug()?.Log("Starting {SpanDetails}", retVal.ToString());
+			_logger.Trace()?.Log("Starting {SpanDetails}", retVal.ToString());
 			return retVal;
 		}
 
 		public void End()
 		{
-			_logger.Debug()?.Log("Ending {SpanDetails}", ToString());
-			if (!Duration.HasValue) Duration = (DateTimeOffset.UtcNow - _start).TotalMilliseconds;
+			if (Duration.HasValue)
+			{
+				_logger.Debug()?.Log("End() called on already ended {Span}. Start time: {Time} (as timestamp: {Timestamp}), Duration: {Duration}ms",
```

Comment:
I think this log sounds misleading... it sounds like you called `End()` already once, which is not the case. This happens when you manually set the duration of the span. If I remember correctly EF Core gives us durations, so we use those instead of calculating ourself. I'd say something like "End() called and duration was already passed to the span manually - no duration calculation. Start time: {Time} (as timestamp: {Timestamp}), Duration: {Duration}ms.:

Reply:
I see... I think we should not allow `End` and `Duration` to get out sync - it will be too error-prone. How about making setter for Duration private and adding another overload for `End` that has duration parameter?

Reply:
Fine by me. This is the example where it makes sense to get the duration from the library and not calculating it ourselves: https://github.com/elastic/apm-agent-dotnet/blob/master/src/Elastic.Apm.EntityFrameworkCore/EfCoreDiagnosticListener.cs#L51

Reply:
None of this is public (it's on on `ISpan`), so no breaking change here.

Reply:
I agree that it's a valid case when duration is determined outside - but OTOH if duration is already know then it means that the span has definitely ended, right? So it makes sense to have API that guides you to do it in one call to minimize chance of setting `Duration` but forgetting to call `End`. Do you agree?

Reply:
Sure, sure. I'm fine with that change.

Reply:
It will require making `Duration` setter private (it was public before). Are you okay with it?

Reply:
It's on an `internal` type, so it's not a breaking change - seems good to me.

Reply:
Unfortunately it was `Duration` setter was `public` on `IExecutionSegment`

Reply:
😬Haven't looked that far up the inheritance chain.. yeah, it was publich :S  In this case I'm not that sure that this was a good idea - sorry for not seeing earlier.

Reply:
Do you think anybody actually used that API? If you think we should revert that change I will do it. I just wanted to make API guide the users into less error prone flow but I can leave with the old API as well.

Reply:
> Do you think anybody actually used that API?

I don't know, but they could and this breaks it, so I'd treat this as a breaking change. If someone used this, they won't be able to compile once they update.

> If you think we should revert

I'm not sure. I agree with your API change. On the other hand, the next version is planned to be `1.0.0`, after the current `1.0.0-beta`, so no major version change and we should avoid breaking changes in such scenarios. I'd personally avoid breaking change now and open an issue to make the change on a major version bump.

But I'm also fine if we keep it as it is now. I leave this up to you.

Reply:
> I'm not sure. I agree with your API change. On the other hand, the next version is planned to be `1.0.0`, after the current `1.0.0-beta`, so no major version change and we should avoid breaking changes in such scenarios. I'd personally avoid breaking change now and open an issue to make the change on a major version bump.

I agree - let's keep breaking changes for when it's really necessary. I'll create a PR shortly to revert and an issue to revisit this for the next major release.

Reply:
#349 - PR to revert.
#350 - issue to revisit this for the next major release when we can slip in breaking changes.

Reply:
BTW should we remove "breaking change" label from this issue?

Reply:
👍 

I also removed the braking change label.